app/models/grda_warehouse/tasks/identify_duplicates.rb

```diff
@@ -16,78 +16,95 @@ def initialize(run_post_processing: true)
-        restore_previously_deleted_destinations
-        Rails.logger.info 'Loading unprocessed clients'
-        started_at = DateTime.now
-        @unprocessed = load_unprocessed
-
-        @dnd_warehouse_data_source = GrdaWarehouse::DataSource.destination.first
-        return unless @dnd_warehouse_data_source
-
-        # compare unprocessed to destinations, looking for a match
-        # If we don't find a match:
-        #   create a new destination (based on the unprocessed client)
-        #   add a associated record to WarehouseClient
-        # If we do find a match:
-        #   create the associated WarehouseClient
-
-        Rails.logger.info "Matching #{@unprocessed.size} unprocessed clients"
-        matched = 0
-        new_created = 0
-        @unprocessed.each_with_index do |c, index|
-          match = check_for_obvious_match(c)
-          client = GrdaWarehouse::Hud::Client.find(c)
-          if match.present?
+      # with_advisory lock with a timeout returns false if the lock was not acquired
+      return if GrdaWarehouseBase.with_advisory_lock('identify_duplicates', timeout_seconds: 0) { identify_duplicates }
+
+      msg = 'Skipping identify duplicates, all ready running.'
+      Rails.logger.warn msg
+      @notifier.ping(msg) if @send_notifications
+    end
+
+    def identify_duplicates
+      restore_previously_deleted_destinations
+      Rails.logger.info 'Loading unprocessed clients'
+      started_at = DateTime.now
+
+      @dnd_warehouse_data_source = GrdaWarehouse::DataSource.destination.first
+      return unless @dnd_warehouse_data_source
+
+      # compare unprocessed to destinations, looking for a match
+      # If we don't find a match:
+      #   create a new destination (based on the unprocessed client)
+      #   add a associated record to WarehouseClient
+      # If we do find a match:
+      #   create the associated WarehouseClient
+
+      Rails.logger.info "Matching #{unprocessed.count} unprocessed clients"
+      matched = 0
+      new_created = 0
+      unprocessed.find_in_batches do |batch|
+        matched_ids = []
+        destination_client_updates = []
+        new_destination_clients = []
+        new_warehouse_clients = []
+        batch.each do |client|
+          match_id = check_for_obvious_match(client)
+          if match_id.present?
             matched += 1
-            destination_client = GrdaWarehouse::Hud::Client.find(match)
-            destination_client.invalidate_service_history
+            matched_ids << match_id
+            destination_client = destination_clients_by_id[match_id]
+
             # Set SSN & DOB if we have it in the incoming client, but not in the destination
             should_save = false
-            if client.DOB.present? && destination_client.DOB.blank?
-              destination_client.DOB = client.DOB
+            if client.DOB.present? && destination_client[:dob].blank?
+              destination_client[:DOB] = client.DOB
```

Comment:
why are you checking `[:dob]` but setting `[:DOB]`?

Reply:
to match data:
```
@destination_clients_by_id ||= all_destination_clients.group_by { |_, _, _, _, dest_id| dest_id }.
         transform_values { |values| values.map { |client| { SSN: client[2], DOB: client[3], id: client[4] } }.first }
```
I think this should be:
```suggestion
            if client.DOB.present? && destination_client[:DOB].blank?
              destination_client[:DOB] = client.DOB
```

Reply:
Yes! Thanks! I discovered that `import` columns don't honor the aliases, so the have to have the DB capitalization, and I missed these!